docs: Snowflake REST Database API support (#136)
localstack/localstack-docs: src/content/docs/snowflake/features/rest-api-endpoints.md

---

> title: REST API

Reviewer: ```suggestion
title: REST APIs
```

> title: REST API Endpoints

Reviewer: Snowflake refers to this feature simply as [REST APIs](https://docs.snowflake.com/en/developer-guide/snowflake-rest-api/snowflake-rest-api), so I think it'd be good to use that instead.

> title: REST API Endpoints

Author: ```suggestion
title: REST API
```

description: Get started with REST API Endpoints in LocalStack for Snowflake
tags: ["Base"]
---

## Introduction

> The [Snowflake REST API](https://docs.snowflake.com/en/developer-guide/snowflake-rest-api/snowflake-rest-api) provides REST API endpoints that allow you to manage schemas and tables in Snowflake. Snowflake REST APIs let you use the programming language of your choice to build your integrations.  

Reviewer: ```suggestion
The [Snowflake REST APIs](https://docs.snowflake.com/en/developer-guide/snowflake-rest-api/snowflake-rest-api) provide endpoints that allow you to manage schemas, tables, and other objects in Snowflake. These APIs enable you to integrate with Snowflake using the programming language of your choice and the known convenience of a [RESTful](https://en.wikipedia.org/wiki/REST) interface.
```

> LocalStack for Snowflake supports REST API endpoints that let you manage your Snowflake data locally. 

Reviewer: ```suggestion
LocalStack for Snowflake supports REST APIs that let you manage your Snowflake data locally. 
```

> ## Supported Snowflake REST API endpoints 

Reviewer: ```suggestion
## Supported Snowflake REST APIs
```

> LocalStack for Snowflake supports REST API Database endpoints that let you manage your Snowflake databases locally. You can create a database, list your databases, or fetch a specific one.

Reviewer 2: please avoid making this section database-specific. we’ll be adding more endpoints in the future that will target other object types (e.g. tables)

Reviewer: Just to maybe add a bit more context for @quetzalliwrites.

The idea of Snowflake REST APIs is to provide a regular RESTful API for many of its resources (like databases, tables, warehouses, roles, etc). In our case it just so happens that until now we've only been able to implement a subset of the endpoints of the Database REST API, but in the future we obviously plan to add implementation of Table REST API, Warehouse REST API, etc., hence why we should try to use generic terms to describe this 🙌

> LocalStack for Snowflake supports the following REST API endpoints to manage your Snowflake data locally:

Reviewer: ```suggestion
LocalStack for Snowflake supports the following REST APIs to manage your Snowflake data locally:
```

> 

Reviewer: Here I think we should add a heading with title: **Database API**

> | Supported Endpoint                                           | Description                                 |

Reviewer: ```suggestion
|Endpoint                                           | Description                                 |
```

> LocalStack for Snowflake supports the following REST API endpoints to manage your Snowflake databases locally:

Reviewer 2: Similar to the above, keep this generic and handle distinctions per object type only in the table.

|---------------------------------------------------|---------------------------------------------|
| `POST /api/v2/databases`                          | Creates a database.                         |
| `GET /api/v2/databases`                           | Lists accessible databases.                 |

> | `GET /api/v2/databases/<name>`                    | Fetch a specific database.                   |

Reviewer: ```suggestion
| `GET /api/v2/databases/<name>`                    | Fetches a specific database.                   |
```

> | `GET /api/v2/databases/<name>`                    | Fetche a specific database.                   |

Reviewer 2: ```suggestion
| `GET /api/v2/databases/<name>`                    | Fetch a specific database.                   |
```
